jubilant/_juju.py

```diff
@@ -837,6 +831,135 @@ def scp(
+        *,
+        reveal: Literal[True],
+        revision: int | None = None,
+        revisions: Literal[False] = False,
+    ) -> Secret[Revealed]: ...
+
+    @overload
+    def show_secret(
+        self,
+        name_or_uri: str | SecretURI,
+        *,
+        reveal: Literal[False] = False,
+        revision: int | None = None,
+        revisions: Literal[False] = False,
+    ) -> Secret[Hidden]: ...
+
+    @overload
+    def show_secret(
+        self,
+        name_or_uri: str | SecretURI,
+        *,
+        reveal: Literal[False] = False,
+        revision: None = None,
+        revisions: Literal[True],
+    ) -> Secret[Hidden]: ...
+
+    def show_secret(
+        self,
+        name_or_uri: str | SecretURI,
+        *,
+        reveal: bool = False,
+        revision: int | None = None,
+        revisions: bool = False,
+    ) -> Secret[Revealed] | Secret[Hidden]:
+        """Get the content of a secret.
+
+        Args:
+            name_or_uri: Name or URI of the secret to reveal.
+            reveal: Whether to reveal the secret content.
+            revisions: Whether to include all revisions of the secret. Mutually
+                exclusive with `reveal` and `revision`.
+            revision: Optional revision number of the secret to reveal. If not specified,
+                the latest revision is revealed.
+        """
+        args = ['show-secret', name_or_uri, '--format', 'json']
+        if reveal:
+            args.append('--reveal')
+        if revisions:
+            args.append('--revisions')
+        if revision is not None:
+            args.extend(['--revision', str(revision)])
+        stdout = self.cli(*args)
+        output: dict[str, SecretResponse] = json.loads(stdout)
+        uri_from_juju, obj = next(iter(output.items()))
+        return Secret[Any](
+            uri=SecretURI('secret:' + uri_from_juju),
+            revision=obj.get('revision', 0),
+            checksum=obj.get('checksum', ''),
+            owner=obj.get('owner', '<model>'),
+            description=obj.get('description', ''),
+            name=obj.get('name', None),
+            created=datetime.datetime.fromisoformat(
+                str(obj.get('created', '').replace('Z', '+00:00'))
+            ),
+            updated=datetime.datetime.fromisoformat(
+                str(obj.get('updated', '').replace('Z', '+00:00'))
+            ),
+            content=obj.get('content', {}).get('Data', None),
+            access=[],
+            revisions=[
+                SecretRevision(
+                    revision=revision.get('revision', ''),
+                    backend=revision.get('backend', ''),
+                    created=datetime.datetime.fromisoformat(
+                        str(revision.get('created', '').replace('Z', '+00:00'))
+                    ),
+                    updated=datetime.datetime.fromisoformat(
+                        str(revision.get('updated', '').replace('Z', '+00:00'))
+                    ),
```

Comment:
Same as above, this can be simplified because they'll always be present and always be a string.